streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoinTest.java

```diff
@@ -135,4 +137,104 @@ public void testJoin() {
 
         processor.checkAndClearProcessResult("2:XX2+YY2", "3:XX3+YY3");
     }
+
+
+    @Test
+    public void testGlobalKTableJoin() throws Exception {
+        StreamsBuilder builder = new StreamsBuilder();
+
+        final int[] expectedKeys = new int[]{0, 1, 2, 3};
+
+        KStream<Integer, String> stream;
+        GlobalKTable<String, String> table; // value of stream optionally contains key of table
+        MockProcessorSupplier<Integer, String> processor;
+        KeyValueMapper<Integer, String, String> keyMapper;
+
+        processor = new MockProcessorSupplier<>();
+        Consumed<Integer, String> streamConsumed = Consumed.with(intSerde, stringSerde);
+        Consumed<String, String> tableConsumed = Consumed.with(stringSerde, stringSerde);
+        stream = builder.stream(topic1, streamConsumed);
+        table = builder.globalTable(topic2, tableConsumed);
+        keyMapper = new KeyValueMapper<Integer, String, String>() {
+            @Override
+            public String apply(Integer key, String value) {
+                final String[] tokens = value.split(",");
+                // Value is comma delimited. If second token is present, it's the key to the global ktable.
+                // If not present, use null to indicate no match
+                return tokens.length > 1 ? tokens[1] : null;
+            }
+        };
+        stream.join(table, keyMapper, MockValueJoiner.TOSTRING_JOINER).process(processor);
+
+        Collection<Set<String>> copartitionGroups = StreamsBuilderTest.getCopartitionedGroups(builder);
+
+        assertEquals("KStream-GlobalKTable joins do not need to be co-partitioned", 0, copartitionGroups.size());
+
+        driver.setUp(builder, stateDir);
+        driver.setTime(0L);
+
+        // push two items to the primary stream. the other table is empty
+
+        for (int i = 0; i < 2; i++) {
+            driver.process(topic1, expectedKeys[i], "X" + expectedKeys[i] + ",FKey" + expectedKeys[i]);
```

Comment:
Nit: can we rename `topic1` -> `streamTopic` and `topic2` -> `globalTableTopic` (below) to make it easier to read the test?

Reply:
good idea